File: typescript/base/src/main/vsdk/toolkit/raycasting/stochasticRaytracing/Path.ts
import { StochasticRaytracingPathNode } from "./StochasticRaytracingPathNode";

/**
A full path, basically an array of 'numberOfNodes' path nodes
*/
export class Path {
  public numberOfNodes: number;
  public nodesAllocated: number;
  public nodes: StochasticRaytracingPathNode[] | null;

  public constructor() {
    this.numberOfNodes = 0;
    this.nodesAllocated = 0;
    this.nodes = null;
  }
}

File: java/base/src/main/java/vsdk/toolkit/raycasting/stochasticRaytracing/Path.java
package vsdk.toolkit.raycasting.stochasticRaytracing;

/**
A full path, basically an array of 'numberOfNodes' path nodes
*/
public class Path {
    public int numberOfNodes;
    public int nodesAllocated;
    public StochasticRaytracingPathNode[] nodes;

    public Path() {
        numberOfNodes = 0;
        nodesAllocated = 0;
        nodes = null;
    }
}
